Make office chairs wheely fun

ItsMeThom opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
Made the rolly office chairs into actual rolling office chairs.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
We should be able to sit and scoot around from desk to desk, or have office jousting matches in the halls when command is asleep. Also its slower than walking (Sprint speed 2.5), but that was never the point.

## Technical details
<!-- Summary of code changes for easier review. -->
Created a new abstract parent `RollingOfficeChairBase` as a vehicle, reparented both wheeled office chairs to it. Kept it in the Wizden folders rather than moving it out to _Goobstation so as not to muck about with the setup of chairs too much.
There is one visual issue; when unanchored and you scoot the chair close to a desk from the North, your legs draw *over* the table, but this is a preexisting issue with all chair masks prior to this change. Im unsure how to fix it.


## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->
https://github.com/user-attachments/assets/2e5f4908-8ee8-421f-a0e1-2837cfb7e707



## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
:cl:
- tweak: Office chairs are mobile now, cubicle workers rejoice!

SolsticeOfTheWinter commented:
I managed to crash my game with this somehow, but it's probably because I was on debug.
LGTM (probably)

SX-7 reviewed (changes requested):
Can't be anchored
You can make the strapped look properly by adding `buckleOffset: "0,-0.05"` to Strap, look at `SeatBase`

SX-7 commented:
> > Can't be anchored You can make the strapped look properly by adding `buckleOffset: "0,-0.05"` to Strap, look at `SeatBase`
> 
> Thanks for the feedback! I'll make these changes after work today. Im not sure if anchorable will work with vehicle but I've not tried it, so if it does, I'll add it in.
> 
> As an aside; you any ideas of how to fix the `renderOver` issues? Im not sure if its coming from RobustToolbox, but in some cases if you rapidly change directions, the chair will overlay the character incorrectly, like if I am travelling south, I will be behind the chair. I've noticed this is an issue on most things based on `vehicle` except for wheelchairs. I think it might be something to do with the `acceleration`

I'm assuming it's because both entities are on the same position, so it's kinda rng which one gets drawn first
I mentioned the buckleOffset, I'm assuming that's the bandaid fix upstream uses to make chair not to that

ItsMeThom commented:
@SX-7 I added the buckleOffset as you suggested and it seems to have solved that issue in *most* cases, theres some some jankness when moving say, east and then tapping north. but thats from upstream and all all vehicles.

I added the anchor component also, and it works fine, no issues anchoring/unanchoring the chairs.
